htap.go

```diff
@@ -1,8 +1,33 @@
 package gosnowflake
 
+import "sync"
+
 type queryContextEntry struct {
 	ID        int   `json:"id"`
 	Timestamp int64 `json:"timestamp"`
 	Priority  int   `json:"priority"`
 	Context   any   `json:"context,omitempty"`
 }
+
+type queryContextCache struct {
+	mutex   *sync.Mutex
+	entries []queryContextEntry
+}
+
+func (qcc *queryContextCache) init() *queryContextCache {
+	qcc.mutex = &sync.Mutex{}
+	return qcc
+}
+
+func (qcc *queryContextCache) add(qce queryContextEntry) {
+	qcc.mutex.Lock()
+	defer qcc.mutex.Unlock()
+
+	qcc.entries = append(qcc.entries, qce)
+}
+
+func (qcc *queryContextCache) addAll(qces []queryContextEntry) {
+	for _, qce := range qces {
+		qcc.add(qce)
```

Comment:
should we lock append for each item in the list or we could do it once for the whole list?

Reply:
This operates on connection objects, which shouldn't be used among threads.

Reply:
my concerns were more about calling multiple times Lock and Unlock (which have its cost?) where Lock and Unlock can be used around for statement

Reply:
I would say, that the overhead here is completely negligible. We shouldn't get here more than 5 elements, and from my testing it's rather one (for non HTAP queries) to two (for HTAP queries).

Reply:
We should think about future cases - now there are only one or two entries but do we know how it will change in the future?

Reply:
No, this works like this. The won't be more than 5 entries. Or to be precise - more than `QUERY_CONTEXT_CACHE_SIZE`, which is 5.

Reply:
Ok, I inlined this method, so I'm resolving this thread as outdated.